src/components/Form/Location/Address.jsx

```diff
@@ -169,17 +117,9 @@ export default class Address extends ValidationElement {
         break
     }
 
-    // Clear the city when moving *from* APO/FPO and *to* something else.
-    if (
```

Comment:
I took out this block of code, it seems like if we are moving from an international address to a domestic one, or vice versa, we would definitely want the city field to be cleared? I understand there are overlaps like Paris (texas, france) or toledo (spain, ohio), but I'm not sure its worth it complicate the code base

Reply:
Love to get some context on this, if anyone knows!

Reply:
APO addresses don't really have a city, but eApp uses the city field to store `APO`, `FPO`, `DPO`. Here's an example from `test7.json`:
```
 "Address": {
                    "type": "location",
                    "props": {
                      "layout": "Address",
                      "street": "No American can study",
                      "city": "APO",
                      "state": "AA",
                      "zipcode": "34003",
                      "country": "POSTOFFICE"
                    }
```
So if you are switching from APO to something else, if you don't clear city, you'll get something odd looking (e.g,. 'APO' listed in the city field).  Without looking at the code carefully, I'm not sure why you don't have to clear the country field for the transition from APO to an international address though.

Speaking generally, one of the pervasive problems across eApp is that it doesn't manage state very well as the applicant optionally enables/disables different options on the form. 

This isn't an existing bug (I think) but just an example:
If I say that I'm married, that will trigger a bunch of fields to be displayed. If I fill out that data, then later switch to saying I've never been married, those fields will get hidden, but that marriage data lingers around, gets saved to the database, re-loaded, etc.

I think this is pretty sloppy and complicates downstream logic a fair bit. If you are toggling between options (in general), all dependent state associated with that option should be cleared. 

However, to play devil's advocate, there might be some sections in the form where there might be a usability case to be made that preserving data allows the user to recover from an errant click.

Reply:
Interesting...I'd like to hear from a more UX or design-centric person (paging @hursey013 and @line47)! I do think it's a valid point, but since the country and state were always getting changed as well, with only the city being preserved, I would err on the side of just blowing it all away.

We don't actually preserve the country value currently, its set to `''` and then reassigned if necessary via a `switch` statement

Reply:
I think allowing for a user to recover from unintentional click is a good idea. I'm also wondering about clearing options when a user navigates away from that page instead of when they toggle between different options.

I think it's odd that `APO` is stored in `city`. 
Why not just have an `APO/FPO/DPO` field instead of using city?

I have a few UI ideas for address updates as well that Ben and I worked on previously. I can add a new issue to show these designs if that would also be helpful.

Reply:
I think it's set up that way because when sending mail to a military address, the `APO|FPO|DPO` replaces the city